### PR TITLE
Fix a crash when calling `copy.copy()` without args

### LINKED ISSUE #8774: no-argument call to copy.copy() crashes
### Bug description

```python
from copy import copy
copy()
```

### Configuration

```ini
n/a
```

### Command used

```shell
pylint a.py
```

### Pylint output

<details open>
    <summary>
        pylint crashed with a ``AstroidError`` and with the following stacktrace:
    </summary>

```
Traceback (most recent call last):
  File "/Users/user/pylint/pylint/lint/pylinter.py", line 788, in _lint_file
    check_astroid_module(module)
  File "/Users/user/pylint/pylint/lint/pylinter.py", line 1017, in check_astroid_module
    retval = self._check_astroid_module(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/pylint/pylint/lint/pylinter.py", line 1063, in _check_astroid_module
    walker.walk(node)
  File "/Users/user/pylint/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/Users/user/pylint/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/Users/user/pylint/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/Users/user/pylint/pylint/checkers/stdlib.py", line 563, in visit_call
    self._check_shallow_copy_environ(node)
  File "/Users/user/pylint/pylint/checkers/stdlib.py", line 514, in _check_shallow_copy_environ
    arg = utils.get_argument_from_call(node, position=0)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/pylint/pylint/checkers/utils.py", line 736, in get_argument_from_call
    raise NoSuchArgumentError
pylint.checkers.utils.NoSuchArgumentError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/user/pylint/pylint/lint/pylinter.py", line 752, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/Users/user/pylint/pylint/lint/pylinter.py", line 790, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

</details>

### Expected behavior

no crash

### Pylint version

```shell
pylint 3.0.0b1
astroid 3.0.0a6-dev0
Python 3.11.2
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Closes #8774

### CODE REVIEW

File: tests/functional/s/shallow_copy_environ.py [hunk 1]
```diff
@@ -27,3 +27,11 @@
...
 
 copy.deepcopy(os.environ)
+
+# Edge cases
+copy.copy()  # [no-value-for-parameter]
+copy.copy(x=test_dict)
+copy.copy(x=os.environ)  # [shallow-copy-environ]
+copy.copy(**{"x": os.environ})  # [shallow-copy-environ]
+copy.copy(**{"y": os.environ})  # [unexpected-keyword-arg]
+copy.copy(y=os.environ)  # [no-value-for-parameter, unexpected-keyword-arg]
```
Comment (jacobtylerwalls) [inline tests/functional/s/shallow_copy_environ.py:37]: I would expect the last two lines to raise the same set of messages. We don't use inference to analyze whether **kwargs contains the missing named parameter `x`. We could.

Comment (jacobtylerwalls) [reply tests/functional/s/shallow_copy_environ.py:37]: Captured in #8785

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM !
